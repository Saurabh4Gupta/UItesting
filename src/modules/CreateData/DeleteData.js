import React from 'react';
import { Modal, Button } from '@dentsu-ui/components/dist/cjs/components/Modal';

const [modalOpen, setModalOpen] = useState(false);
const DeleteData = () => {
  const handleDelete = () => {
    console.log('Hello deleted');
    setModalOpen(false);
  }
  return (
    <>
      <Modal isOpen={modalOpen} onClose={() => setModalOpen(false)}>
        <Modal.Header hasCloseButton title="Delete this data requests" />
        <Modal.Body>
          <p>
            Are you sure want to delete this request?
          </p>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={() => setModalOpen(false)}>
            cancel
          </Button>
          <Button onClick={handleDelete}>
            Send
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  )
}
export default DeleteData;
