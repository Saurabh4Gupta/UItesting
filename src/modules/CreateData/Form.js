/* eslint-disable react/prop-types */
import React, { useState } from 'react';

import {
  Select,
  FormField,
  TextInput,
  Textarea,
  DateInput,
  Dropzone,
  Stack,
  Box,
} from '@dentsu-ui/components';
import constant from '../../utils/constant';

const Form = (props) => {
  const { MAX_FILE_SIZE, ALLOWED_FILE_TYPES } = constant;
  const { values, handleChange, handleSelectField, errors, cmsData, options, monthOptions, dueDate } = props;
  const [files, setFiles] = useState([]);
  const handleInit = () => {
    console.log('Dropzone instance has initialised', files);
  };
  return (
    <>
      <FormField
        {...(errors.localMarket ? { error: errors.localMarket } : {})}
        label={cmsData.localMarketLabel}
        hint={cmsData.localMarketHint}
      >
        <Select
          placeholder={cmsData.selectPlaceHolder}
          name="localMarket"
          options={options}
          onChange={(selected, event) => handleSelectField(selected, event)}
          value={
            options.find(key => key.value === values.localMarket.value)
          }
        />
      </FormField>
      <FormField
        {...(errors.name ? { error: errors.name } : {})}
        label={cmsData.nameLabel}
        hint={cmsData.nameHint}
      >
        <TextInput
          name="name"
          maxLength={255}
          value={values.name}
          onChange={handleChange}
        />
      </FormField>
      <FormField
        label={cmsData.briefLabel}
        {...(errors.briefing ? { error: errors.briefing } : {})}
        hint={cmsData.briefHint}
      >
        <Textarea
          maxLength={4000}
          isFullWidth="true"
          placeholder=""
          name="briefing"
          value={values.briefing}
          onChange={handleChange}
        />
      </FormField>

      <FormField
        label={cmsData.templateFileLabel}
        hint={cmsData.templateFileHint}
        {...(errors.actualData ? { error: errors.actualData } : {})}
      >
        <Dropzone
          allowMultiple={false}
          allowReorder
          onInit={() => handleInit()}
          onUpdateFiles={(fileItems) => {
            setFiles(fileItems);
          }}
          maxFiles={1}
          maxFileSize={MAX_FILE_SIZE}
          server="./"
          acceptedFileTypes={ALLOWED_FILE_TYPES}
          labelMaxFileSizeExceeded={cmsData.uploadFileLargeMessage}
          labelFileTypeNotAllowed={cmsData.labelFileTypeNotAllowed}
        />

      </FormField>
      <Stack flex="row">
        <Box mr="30px">
          <FormField
            label={cmsData.actualDataLabel}
            {...(errors.actualData ? { error: errors.actualData } : {})}
            hint={cmsData.actualDataHint}
          >
            <Select
              name="actualData"
              options={monthOptions}
              value={values.actualData}
              placeholder={cmsData.selectPlaceHolder}
              onChange={(selected, event) => {
                handleSelectField(selected, event)
              }}
            />
          </FormField>
        </Box>
        <Box>
          <FormField
            label={cmsData.forecastDataLabel}
            {...(errors.forecastData ? { error: errors.forecastData } : {})}
            hint={cmsData.actualDataHint}
          >
            <Select
              name="forecastData"
              placeholder={cmsData.selectPlaceHolder}
              value={values.forecastData}
              options={monthOptions}
              onChange={(selected, event) => {
                handleSelectField(selected, event)
              }}
            />

          </FormField>
        </Box>
      </Stack>
      <FormField
        {...(errors.dueDate ? { error: errors.dueDate } : {})}
        label={cmsData.dueDateLabel}
        hint={cmsData.dueDateHint}
      >
        <DateInput
          placeholder={cmsData.datePlaceHolder}
          dateFormat="DD/MM/YYYY"
          parseDate={(date) => new Date(date)}
          formatDate={(date) => date.toLocaleDateString()}
          onChange={(date) => handleSelectField(date, { name: 'dueDate' })}
          minDate={new Date()}
          value={dueDate}
        />
      </FormField>

      <FormField
        label={cmsData.assignToLabel}
        {...(errors.assignTo ? { error: errors.assignTo } : {})}
        hint={cmsData.assignToHint}
      >
        <Select
          menuPosition="fixed"
          options={options}
          name="assignTo"
          placeholder={cmsData.selectPlaceHolder}
          onChange={(selected, event) => handleSelectField(selected, event)}
          value={
            options.find(key => key.value === values.assignTo.value)
          }
        />
      </FormField>
    </>
  )
};

export default Form;
