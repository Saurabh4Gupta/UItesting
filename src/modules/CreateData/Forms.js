import React from 'react';

const Form = () => <h1>Hello</h1>
export default Form;
