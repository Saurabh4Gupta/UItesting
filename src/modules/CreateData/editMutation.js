import gql from 'graphql-tag';


const EDIT_DATA_REQUEST = gql`
mutation EditDataRequest($data: createDataRequestInput!){
  editDataRequest(data: $data){
    message
    status
    data {
      id
      name
      briefing
      actualData
      forecastData
      reportingYear
      dueDate
      owners
      isComplete
      filename
      blobId
    }
  }
}`;

export default EDIT_DATA_REQUEST;
