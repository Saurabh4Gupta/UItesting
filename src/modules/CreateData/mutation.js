import { gql } from '@apollo/client';

const CREATE_DATA_REQUEST = gql`
  mutation CreateDataRequest($data: createDataRequestInput!) {
    createDataRequests(data: $data) {
      message
      status
      data {
        id
      }
    }
  }
`;

export default CREATE_DATA_REQUEST;
