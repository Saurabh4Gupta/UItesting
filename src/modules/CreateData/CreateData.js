import React, { useState, useEffect, useContext } from 'react'
import { useQuery } from '@apollo/client';
import { Caption, Subheading, TextContainer, Button, Stack, Modal } from '@dentsu-ui/components';
import PropTypes from 'prop-types';
import Form from './Form'
import useCustomForm from '../../hooks/useCustomForm';
import validationRule from '../../utils/validate';
import { monthOptions, reportingYear } from '../Mock/mockData'
import { GET_USERS } from './queries';
import { MarketOptionsContext } from '../../contexts/marketOptions';

const CreateData = (props) => {
  const [userRequest, setUserRequest] = useState([]);
  const { loading, error, data } = useQuery(GET_USERS);
  const { cmsData, market, isModalOpen, handleModal, addRequest } = props;
  const [isReadyToSubmit, setIsReadyToSubmit] = useState(false);
  const [isLoading, setLoading] = useState(false);
  const initialValues = {
    localMarket: market,
    name: '',
    briefing: '',
    reportingYear: '',
    actualData: '',
    forecastData: '',
    dueDate: '',
    assignTo: '',
  };
  const { handleChange, values, forecastOptions,
    handleSelectField, handleSubmit,
    errors, handleCancel } = useCustomForm({ initialValues, validate: validationRule });

  useEffect(() => {
    const isAnyValidationError = errors && !!(errors.localMarket || errors.name
      || errors.briefing || errors.dueDate || errors.assignTo || errors.forecastData
      || errors.forecastData || errors.reportingYear);
    const isAllValuesFilled = values.localMarket && values.name && values.assignTo
      && values.dueDate && values.forecastData && values.actualData && values.briefing && values.reportingYear;
    setIsReadyToSubmit(isAllValuesFilled && !isAnyValidationError);
  }, [errors, values]);

  useEffect(() => {
    handleChange({ target: { name: 'localMarket', value: market } });
  }, [market]);

  // eslint-disable-next-line consistent-return
  useEffect(() => {
    if (loading) {
      setLoading(false)
    }
    if (error) return `Error! ${error}`;
    if (data) {
      setUserRequest(data)
    }
  }, [data, loading, error]);
  const closeModalHandler = () => {
    handleModal(false)
    handleCancel();
  }
  const onSubmit = async () => {
    handleSubmit();
    if (isReadyToSubmit) {
      setLoading(true);
      // mutation will be done here
      setTimeout(() => {
        setLoading(false);
        closeModalHandler();
        addRequest(values)
      }, 1000);
    }
  }
  const handleCreateData = () => {
    handleModal(true);
  };

  const marketOptions = useContext(MarketOptionsContext);
  const formMarketOption = marketOptions.filter(item => item.value !== '');

  return (
    <>
      <Modal isOpen={isModalOpen} onClose={closeModalHandler}>
        <Modal.Header hasCloseButton title={cmsData.createNewDataRequest} />
        <Modal.Body>
          <Form
            values={values}
            handleChange={handleChange}
            handleSelectField={handleSelectField}
            errors={errors}
            cmsData={cmsData}
            options={formMarketOption}
            monthOptions={monthOptions}
            reportingYear={reportingYear}
            forecastOptions={forecastOptions}
            userList={userRequest.getUsers}
          />
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={closeModalHandler}>
            {cmsData.cancel}
          </Button>
          <Button isLoading={isLoading} onClick={onSubmit}>{cmsData.create}</Button>
        </Modal.Footer>
      </Modal>
      <Stack flexDirection="row" justifyContent="space-between">
        <Stack>
          <TextContainer>
            <Subheading>{cmsData.productivityDatarequestHeading}</Subheading>
            <Caption isAssistive>
              {cmsData.productivityDatarequestCaption}
            </Caption>
          </TextContainer>
        </Stack>
        <Button variant="secondary" iconLeft="add" onClick={handleCreateData}>
          {cmsData.createNewDataRequest}
        </Button>
      </Stack>
    </>
  )
};
CreateData.propTypes = {
  cmsData: PropTypes.object,
  market: PropTypes.object,
  isModalOpen: PropTypes.bool,
  handleModal: PropTypes.func,
  addRequest: PropTypes.func,
}
CreateData.defaultProps = {
  cmsData: {},
  market: 'UK',
  isModalOpen: false,
  handleModal: () => { },
  addRequest: () => { },
}

export default CreateData;
