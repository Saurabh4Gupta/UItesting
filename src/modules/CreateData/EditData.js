import React, { useState, useEffect } from 'react'
import { Button, Modal } from '@dentsu-ui/components';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import Form from './Form'
import useCustomForm from '../../hooks/useCustomForm';
import validationRule from '../../utils/validate';
import { options, monthOptions, reportingYear } from '../Mock/mockData'

const EditData = (props) => {
  const { cmsData, isModalOpen, handleModal } = props;
  const [isReadyToSubmit, setIsReadyToSubmit] = useState(false);
  const initialValues = {
    localMarket: { value: 'UK', label: 'United Kingdom' },
    name: 'Productivity Q2 2020',
    briefing: 'Give some details about this quarter',
    reportingYear:{ value: 'April 2021 - March 2022', label: 'April 2021 - March 2022' },
    actualData: { value: 0, label: '0 months' },
    forecastData: { value: 12, label: '12 months' },
    dueDate: new Date(),
    assignTo: { value: 'UK', label: 'United Kingdom' },
  };
  const { handleChange, values, forecastOptions,
    handleSelectField, handleSubmit,
    errors, handleCancel } = useCustomForm({ initialValues, validate: validationRule });

  useEffect(() => {
    const isAnyValidationError = errors && !!(errors.localMarket || errors.name
      || errors.briefing || errors.dueDate || errors.assignTo || errors.forecastData
      || errors.forecastData || errors.reportingYear);
    const isAllValuesFilled = values.localMarket && values.name && values.assignTo
      && values.dueDate && values.forecastData && values.actualData && values.briefing && values.reportingYear;
    setIsReadyToSubmit(isAllValuesFilled && !isAnyValidationError);
  }, [errors, values]);
  const closeModalHandler = () => {
    handleModal(false)
    handleCancel();
  }
  const onSubmit = () => {
    handleSubmit();
    if (isReadyToSubmit) {
      // console.log('??????????', values);
    closeModalHandler()
    // updateData(values)
    }
  }
  return (
    <>
      <Modal isOpen={isModalOpen} onClose={closeModalHandler}>
        <Modal.Header hasCloseButton title={cmsData.editDataRequest} />
        <Modal.Body>
          <Form
            values={values}
            handleChange={handleChange}
            handleSelectField={handleSelectField}
            errors={errors}
            cmsData={cmsData}
            options={options}
            monthOptions={monthOptions}
            dueDate={values.dueDate}
            reportingYear={reportingYear}
            forecastOptions={forecastOptions}
          />
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={closeModalHandler}>
            {cmsData.cancel}
          </Button>
          <Button onClick={onSubmit}>{cmsData.save}</Button>
        </Modal.Footer>
      </Modal>
    </>
)
};
EditData.propTypes = {
    cmsData: PropTypes.object,
    isModalOpen: PropTypes.bool,
    handleModal: PropTypes.func,
  }
  EditData.defaultProps = {
    cmsData: {},
    isModalOpen: false,
    handleModal: () => { },
  }
export default withRouter(EditData);
