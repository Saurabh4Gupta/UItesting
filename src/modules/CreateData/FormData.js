/* eslint-disable operator-linebreak */
import React, { useState, useEffect, useContext } from 'react';
import { useMutation, useQuery } from '@apollo/client';
import { Button, Modal } from '@dentsu-ui/components';
import PropTypes from 'prop-types';

import Toast from '@dentsu-ui/components/dist/cjs/components/Toast';
import moment from 'moment';
import Form from './Form';
import useCustomForm from '../../hooks/useCustomForm';
import { checkValidation, validationRule } from '../../utils/validate';
import { monthOptions, reportingYearData } from '../Mock/mockData';

import { GET_USERS } from './queries';
import { MetaDataContext } from '../../contexts/marketOptions';
import FILE_UPLOAD from '../FileUpload/mutation';
import CREATE_DATA_REQUEST from './mutation';
import { getInitialValues } from '../../utils/formValues';

const toast = Toast();
const CreateData = (props) => {
  const {
    cmsData,
    market,
    isModalOpen,
    handleModal,
    refetch,
    setMarket,
    isEdit,
    prodRequest,
  } = props;

  const [isReadyToSubmit, setIsReadyToSubmit] = useState(false);

  const [uploadFile, { loading: fileLoading, error: fileError }] = useMutation(
    FILE_UPLOAD,
  );
  const [createDataRequest, { loading, error, data: createData }] = useMutation(
    CREATE_DATA_REQUEST,
  );
  const [userData, setUserData] = useState([]);

  // const [files, setFiles] = useState([]);
  const { loading: userLoading, error: userError, data: userList } = useQuery(
    GET_USERS,
  );

  const initialValues = getInitialValues({
    isEdit,
    market,
    data: prodRequest || null,
  });
  const {
    handleChange,
    values,
    handleSelectField,
    handleSubmit,
    errors,
    handleCancel,
    handleOwners,
    setErrors,
    setValues,
    forecastOptions,
  } = useCustomForm({ initialValues, validate: validationRule });

  useEffect(() => {
    if (userList) {
      const { data } = userList.getUsers;
      setUserData(data);
    }
  }, [userList]);

  const closeModalHandler = () => {
    handleModal(false);
    handleCancel();
  };

  useEffect(() => {
    if (createData) {
      const { status } = createData.createDataRequests;
      if (status === 200) {
        setMarket(values.localMarket);
        closeModalHandler();
        refetch();
        return toast({
          title: cmsData.toastRequestCreated,
          status: 'success',
        });
      }
    }
  }, [createData]);

  useEffect(() => {
    const { isAllValuesFilled, isAnyValidationError } = checkValidation(
      values,
      errors,
    );
    setIsReadyToSubmit(isAllValuesFilled && !isAnyValidationError);
  }, [errors, values]);

  useEffect(() => {
    if (fileError) {
      toast({
        title: '',
        content: fileError.message,
        status: 'danger',
      });
    }
  }, [fileError, error, userError]);

  useEffect(() => {
    handleChange({ target: { name: 'localMarket', value: market } });
  }, [market]);

  /* parsing file to base64 */

  const handleFile = (fileItems) => {
    if (fileItems.length === 0) {
      setValues((prevState) => ({
        ...prevState,
        file: [],
      }));
      return;
    }
    const reader = new FileReader();
    const { file } = fileItems[0];
    reader.readAsDataURL(file);
    reader.onloadend = async () => {
      const fileUrl = reader.result;
      setValues((prevState) => ({
        ...prevState,
        file: { fileUrl, filename: file.name, size: file.size },
      }));
      setErrors((prevState) => ({
        ...prevState,
        file: '',
      }));
    };
  };

  /* On submit */

  async function onSubmit() {
    handleSubmit();
    if (isReadyToSubmit) {
      const { file } = values;
      if (isEdit) {
        // Write mutation here check below for
        // reference and in last return the  toast with status 200
      }
      const { data: uploadData } = await uploadFile({ variables: { file } });
      const { data: fileData, status } = uploadData.uploadFile;
      if (status !== 200) {
        setErrors((prevState) => ({
          ...prevState,
          file: cmsData.fileUploadFailed,
        }));
        return;
      }
      const {
        localMarket,
        name,
        briefing,
        actualData,
        forecastData,
        dueDate,
        assignTo,
        // eslint-disable-next-line no-shadow
        reportingYear,
      } = values;

      const reqData = {
        overviewId: localMarket.overviewId,
        blobId: fileData.blobId,
        actualData,
        forecastData,
        name,
        briefing,
        reportingYear: moment(reportingYear.value).format('YYYY'),
        dueDate,
        owners: assignTo,
        filename: fileData.filename,
      };

      createDataRequest({ variables: { data: reqData } });
    }
  }
  const { marketOptions } = useContext(MetaDataContext);
  const formMarketOption = marketOptions.filter((item) => item.value !== 'All');

  return (
    <>
      <Modal isOpen={isModalOpen} onClose={closeModalHandler}>
        <Modal.Header hasCloseButton title={cmsData.createNewDataRequest} />
        <Modal.Body>
          <Form
            values={values}
            handleChange={handleChange}
            handleSelectField={handleSelectField}
            errors={errors}
            cmsData={cmsData}
            options={formMarketOption}
            monthOptions={monthOptions}
            reportingYear={reportingYearData}
            handleOwners={handleOwners}
            setFiles={handleFile}
            userList={userData}
            loading={userLoading}
            forecastOptions={forecastOptions}
            isEdit={isEdit}
          />
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={closeModalHandler}>
            {cmsData.cancel}
          </Button>
          <Button isLoading={loading || fileLoading} onClick={onSubmit}>
            {cmsData.create}
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
};
CreateData.propTypes = {
  cmsData: PropTypes.object,
  market: PropTypes.object,
  isModalOpen: PropTypes.bool,
  handleModal: PropTypes.func,
  refetch: PropTypes.func,
  setMarket: PropTypes.func,
  isEdit: PropTypes.bool,
  prodRequest: PropTypes.object,
};
CreateData.defaultProps = {
  cmsData: {},
  market: 'UK',
  isModalOpen: false,
  handleModal: () => {},
  refetch: () => {},
  setMarket: () => {},
  isEdit: false,
  prodRequest: {},
};

export default CreateData;
