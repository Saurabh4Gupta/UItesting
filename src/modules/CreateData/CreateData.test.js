import React from 'react';
import { shallow } from 'enzyme';
import CreateData from './CreateData';

describe('Success test cases for: Loader component', () => {
  let wrapper;

  describe('Snapshot test for  Loader component', () => {
    beforeEach(() => {
      wrapper = shallow(<CreateData />);
    });

    it('Test to match the snapsot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Renders correctly', () => {
      expect(wrapper.exists()).toBe(true);
    });

    it('handle create ', () => {
      wrapper.find("[data-test='handle-create']").simulate('click')
    })

    it('handle model', () => {
      wrapper.find("[data-test='handle-modal']").simulate('click')
    })

    it('handle create', () => {
      wrapper.find("[data-test='handle-button']").simulate('click')
    })
  });
});

describe('Failure test cases for: Duplicate data page', () => { });
