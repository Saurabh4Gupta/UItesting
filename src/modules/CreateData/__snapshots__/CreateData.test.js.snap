// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success test cases for: Loader component Snapshot test for Loader component Test to match the snapsot 1`] = `
<Fragment>
  <ForwardRef
    isOpen={false}
    onClose={[Function]}
  >
    <ForwardRef
      hasCloseButton={true}
      title="Create new data request"
    />
    <ForwardRef>
      <Form
        cmsData={
          Object {
            "actual": "actuals",
            "actualCapitalized": "Actual",
            "actualDataHint": "Number of months of actual productivity data",
            "actualDataLabel": "Actual Data",
            "assignToHint": "Only the person or team you assign this to will have access. Once you click 'Create' ,
  they will be sent email notifications asking them to fill in the data request.",
            "assignToLabel": "Assign To",
            "assignedTo": "Assigned to",
            "briefHint": "Tell us in detail what this data request is for, max 4000 characters",
            "briefLabel": "Briefing",
            "cancel": "Cancel",
            "close": "Close",
            "complete": "Complete",
            "completeLabel": "Complete",
            "create": "Create",
            "createNewDataRequest": "Create new data request",
            "datePlaceHolder": "Date",
            "delete": "Delete",
            "dueBy": "Due by",
            "dueDateHint": "Enter the date that you need this data request by",
            "dueDateLabel": "Due Date",
            "edit": "Edit",
            "editDataRequest": "Edit data request",
            "editRequest": "Edit request",
            "emptyCompletedProductivityDatarequestCaption": "No Productivity data requests have been moved to this section yet.",
            "emptyLabel": "Create new data request",
            "emptyProductivityDatarequestCaption": "There are currently no active Productivity data requests",
            "forecast": "forecast",
            "forecastCapitalized": "Forecast",
            "forecastDataHint": "Number of months of forecasted productivity data",
            "forecastDataLabel": "Forecast Data",
            "labelFileTypeNotAllowed": "File must be of type .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx",
            "lastUpdate": "Last Update",
            "localMarketHint": "Choose the market you wish to create this data request for",
            "localMarketLabel": "Local Market",
            "moveToComplete": "Move to Complete",
            "moveToOnGoing": "Move to Ongoing",
            "nameHint": "Make sure it's clear to help others identify it.Max 255 characters",
            "nameLabel": "Data request name",
            "ongoing": "Ongoing",
            "ongoingLabel": "Ongoing",
            "overdue": "Overdue",
            "productivityDataRequested": "Productivity data requested",
            "productivityDatarequestCaption": " The ongoing and complete productivity data requests created for this client",
            "productivityDatarequestHeading": "Productivity data requests",
            "productivityRequest": "Productivity Request",
            "reportingYear": "Reporting year",
            "reportingYearHint": "The reporting year this data request is for",
            "reportingYearLabel": "Reporting year",
            "save": "Save",
            "selectPlaceHolder": "Please select",
            "showMore": "Show more",
            "templateFileHint": "Only .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx files in 20mb or less",
            "templateFileLabel": "Add tracker template file",
            "trackerTemplate": "Tracker template",
            "upload": "Upload",
            "uploadButtonText": "Upload new file",
            "uploadFileLabel": "To upload a new file,drag and drop you file below or browse using the link",
            "uploadFileLargeMessage": "File must be less than 20MB",
            "viewClientsCaption": "Choose the client you wish to view",
            "viewClientsHeading": "My clients",
            "viewDetails": "View Details",
          }
        }
        errors={Object {}}
        handleChange={[Function]}
        handleSelectField={[Function]}
        monthOptions={
          Array [
            Object {
              "label": "0 months",
              "value": 0,
            },
            Object {
              "label": "1 month",
              "value": 1,
            },
            Object {
              "label": "2 months",
              "value": 2,
            },
            Object {
              "label": "3 months",
              "value": 3,
            },
            Object {
              "label": "4 months",
              "value": 4,
            },
            Object {
              "label": "5 months",
              "value": 5,
            },
            Object {
              "label": "6 months",
              "value": 6,
            },
            Object {
              "label": "7 months",
              "value": 7,
            },
            Object {
              "label": "8 months",
              "value": 8,
            },
            Object {
              "label": "9 months",
              "value": 9,
            },
            Object {
              "label": "10 months",
              "value": 10,
            },
            Object {
              "label": "11 months",
              "value": 11,
            },
            Object {
              "label": "12 months",
              "value": 12,
            },
          ]
        }
        options={
          Array [
            Object {
              "label": "Argentina",
              "value": "ARG",
            },
            Object {
              "label": "Australia",
              "value": "AUS",
            },
            Object {
              "label": "Belgium",
              "value": "BEL",
            },
            Object {
              "label": "United Kingdom",
              "value": "UK",
            },
            Object {
              "label": "USA",
              "value": "USA",
            },
            Object {
              "label": "France",
              "value": "FR",
            },
            Object {
              "label": "New Zealand",
              "value": "NZ",
            },
            Object {
              "label": "Singapore",
              "value": "SGP",
            },
            Object {
              "label": "Turkey",
              "value": "Tk",
            },
            Object {
              "label": "Italy",
              "value": "IT",
            },
          ]
        }
        reportingYear={
          Array [
            Object {
              "label": "April 2021  -  March 2022",
              "value": "April 2021  -  March 2022",
            },
            Object {
              "label": "April 2020  -  March 2021",
              "value": "April 2020  -  March 2021",
            },
            Object {
              "label": "April 2019  -  March 2020",
              "value": "April 2019  -  March 2020",
            },
          ]
        }
        values={
          Object {
            "actualData": "",
            "assignTo": "",
            "briefing": "",
            "dueDate": "",
            "forecastData": "",
            "localMarket": "UK",
            "name": "",
            "reportingYear": "",
          }
        }
      />
    </ForwardRef>
    <ForwardRef>
      <ForwardRef
        onClick={[Function]}
        variant="secondary"
      >
        Cancel
      </ForwardRef>
      <ForwardRef
        isLoading={false}
        onClick={[Function]}
      >
        Create
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
  <Stack
    flexDirection="row"
    justifyContent="space-between"
  >
    <Stack>
      <ForwardRef>
        <Subheading>
          Productivity data requests
        </Subheading>
        <ForwardRef
          isAssistive={true}
        >
           The ongoing and complete productivity data requests created for this client
        </ForwardRef>
      </ForwardRef>
    </Stack>
    <ForwardRef
      iconLeft="add"
      onClick={[Function]}
      variant="secondary"
    >
      Create new data request
    </ForwardRef>
  </Stack>
</Fragment>
`;
